src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -67,14 +68,14 @@ ov::npuw::JustInferRequest::JustInferRequest(const std::shared_ptr<ov::npuw::Com
                     // Note: these buffers are allocated to the entire NWAY (> tail_size)
                     for (auto&& p : proto_comp_model_desc.spatial->params) {
                         const auto& iport = proto_comp_model_desc.compiled_model->inputs()[p.idx];
-                        m_spatial_io[real_idx].input_tails[p.idx] =
-                            ov::get_tensor_impl(ov::Tensor(iport.get_element_type(), iport.get_shape()));
+                        m_spatial_io[real_idx].input_tails[p.idx] = ov::get_tensor_impl(
+                            allocTensor(iport.get_element_type(), iport.get_shape(), *proto_comp_model_desc.device_it));
```

Comment:
just thinking.. if you're using allocTensor only where we store ITensors, why can't `allocTensor` return the `ITensor` so you don't need to call `get_tensor_impl` everywhere?

Reply:
Done